extensions/schema-compare/src/dialogs/schemaCompareDialog.ts

```diff
@@ -147,20 +167,52 @@ export class SchemaCompareDialog {
 				serverName: targetServerDropdownValue.name,
 				databaseName: this.targetDatabaseDropdown.value.toString(),
 				ownerUri: ownerUri,
+				projectFilePath: '',
+				folderStructure: '',
+				targetScripts: [],
 				packageFilePath: '',
+				dataSchemaProvider: '',
 				connectionDetails: undefined,
-				connectionName: targetServerDropdownValue.connection.options.connectionName,
+				connectionName: targetServerDropdownValue.connection.options.connectionName
+			};
+		} else if (this.targetIsDacpac) {
+			this.schemaCompareMainWindow.targetEndpointInfo = {
+				endpointType: mssql.SchemaCompareEndpointType.Dacpac,
+				serverDisplayName: '',
+				serverName: '',
+				databaseName: '',
+				ownerUri: '',
 				projectFilePath: '',
 				folderStructure: '',
 				targetScripts: [],
-				dataSchemaProvider: ''
+				dataSchemaProvider: '',
+				packageFilePath: this.targetTextBox.value,
+				connectionDetails: undefined
+			};
+		} else {
+			this.schemaCompareMainWindow.targetEndpointInfo = {
+				endpointType: mssql.SchemaCompareEndpointType.Project,
+				projectFilePath: this.targetTextBox.value,
+				folderStructure: this.targetStructureDropdown!.value as string,
+				targetScripts: await this.getTargetScripts(false),
+				dataSchemaProvider: await this.getDsp(false),
+				serverDisplayName: '',
+				serverName: '',
+				databaseName: '',
+				ownerUri: '',
+				packageFilePath: '',
+				connectionDetails: undefined
 			};
 		}
 
 		TelemetryReporter.createActionEvent(TelemetryViews.SchemaCompareDialog, 'SchemaCompareStart')
 			.withAdditionalProperties({
 				sourceIsDacpac: this.sourceIsDacpac.toString(),
-				targetIsDacpac: this.targetIsDacpac.toString()
+				targetIsDacpac: this.targetIsDacpac.toString(),
+				sourceIsDatabase: this.sourceIsDatabase.toString(),
+				targetIsDatabase: this.targetIsDatabase.toString(),
+				sourceIsProject: this.sourceIsProject.toString(),
+				targetIsProject: this.targetIsProject.toString()
```

Comment:
instead of having 6 variables to keep track of, can there just be an enum with dacpac, database, and project that a source and target variable can be set to?